src/cpp/rtps/builtin/discovery/participant/PDPClient.cpp

```diff
@@ -769,9 +769,17 @@ void PDPClient::announceParticipantState(
                         // if we are matched to a server report demise
                         if (svr.is_connected)
                         {
+                            GuidPrefix_t srv_guid_prefix = svr.guidPrefix;
+
+                            if (getRTPSParticipant()->is_secure())
+                            {
+                                // Need the mangled guid prefix in this case
+                                srv_guid_prefix = get_participant_proxy_data(svr.guidPrefix)->m_guid.guidPrefix;
+                            }
```

Comment:
Based on the CI results, it looks like this should be used with security guards.